src/cryptonote_protocol/cryptonote_protocol_handler.h

```diff
@@ -113,12 +113,22 @@ namespace cryptonote
     const block_queue &get_block_queue() const { return m_block_queue; }
     void stop();
     void on_connection_close(cryptonote_connection_context &context);
-    void set_max_out_peers(unsigned int max) { m_max_out_peers = max; }
+    void set_max_out_peers(epee::net_utils::zone zone, unsigned int max) { m_max_out_peers[zone] = max; }
+    unsigned int get_max_out_peers(epee::net_utils::zone zone) const
+    {
+      const auto it = m_max_out_peers.find(zone);
+      if (it == m_max_out_peers.end())
+      {
+        MWARNING(epee::net_utils::zone_to_string(zone) << " max out peers not set, using default");
```

Comment:
Do you really want to spam people who set no limit ? I assume this'd do. It maybe a rare case, but it's legitimate. Or is a max always set even if the user doesn't set one ?

Reply:
Maybe I'm missing something but I can't see how it wouldn't get set by a regular user; it should always get set in here even if a user sets a limit of 0 or doesn't include the `--out-peers` flag: https://github.com/j-berman/monero/blob/299fbd007687534a3b5abb87c2b3f2f596ecdacb/src/p2p/net_node.inl#L388

The only way I believe it wouldn't get set is if a dev is doing something custom where they skip setting it before calling a function that reads it, like in the unit test I updated in this PR (which was hanging before I made this change). I.e. it should always be set explicitly before it's read in normal use.

Maybe there's a better way to handle this? Default to 0 in case it's not set, and maybe only warn once?

Reply:
I did not look. If it's impossible with "normal" usage, it's fine by me.

Reply:
I did not look. If it's impossible with "normal" usage, it's fine by me.